Dockerfile

```
@@ -6,9 +6,10 @@ ENV PYTHONUNBUFFERED 1
 
 WORKDIR /code
 
-COPY ./requirements.txt /code/requirements.txt
+COPY ./Pipfile* /code
 RUN python3 -m pip install --upgrade pip \
-    && pip install --no-cache-dir -r requirements.txt
+    && pipenv lock --keep-outdated --requirements > requirements.txt
+    && pip install -r requirements.txt
```

We can use pipenv directly here.
```suggestion
RUN pip install --upgrade --no-cache-dir pip pipenv \
    && pipenv install --dev --system --deploy \
    && pip uninstall -y pipenv virtualenv-clone virtualenv
```

---

Any reason we are uninstalling virtualenv-clone and virtualenv?

---

Assuming pipenv or poetry installs them.